src/main/java/br/unicamp/cst/util/MindViewer.java

```diff
@@ -107,10 +127,16 @@ public MindViewer(Mind mind, String windowName, List<Codelet> behavioralCodelets
     
     
     private void buildMindModulePanels(Mind mind){
-        if(mind.getMotivationalSubsystemModule().verifyExistCodelets()){
-            motivationalModuleViewer = new MotivationalSubsystemViewer(Long.parseLong(txtRefreshTime.getText()), mind);
-            tbModules.add("Motivational Subsystem", motivationalModuleViewer);
-        }
+        
+        //if(mind.getMotivationalSubsystemModule().verifyExistCodelets()){
+        if(mind.getGroupsNumber() > 0) {
+            if (mind.getGroupList("Motivational") != null) {
+                if (mind.getGroupList("Motivational").size() > 0) {
```

Comment:
There are three if statements enclosed that could be merged into a single one using `&&`operator.